### PR TITLE
Native inbox compose

### PR DESCRIPTION
**Pull request by domonkosadam**
refs: MBL-14853
affects: Student, Teacher
release note: none

test plan:
- test inbox / new message

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/CoreTests/Inbox/ComposeMessage/ComposeMessageViewModelTests.swift:59:1: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Inbox/Model/Entities/InboxCourse.swift:62:44: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (colon)
    /Users/vagrant/git/Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:34:11: error: Comment Spacing Violation: Prefer at least one space after slashes for comments (comment_spacing)
    ❌ /Users/vagrant/git/Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:122:43: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration (opening_brace)
    /Users/vagrant/git/Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:108:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:167:73: error: Unused Closure Parameter Violation: Unused parameter in a closure should be replaced with _ (unused_closure_parameter)
    ❌ /Users/vagrant/git/Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:186:44: Unused Closure Parameter Violation: Unused parameter in a closure should be replaced with _ (unused_closure_parameter)
    
    
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c9a0fac162fd4f638c78425c94323653df8e9353
</p>

File: TestsFoundation/TestsFoundation.xcodeproj/project.pbxproj
Comment (vargaat) [inline TestsFoundation/TestsFoundation.xcodeproj/project.pbxproj:1]: Unnecessary changes.

**Review by vargaat — CHANGES_REQUESTED** [review]
- Let's add tests for files that we think are somewhat final.
- Adhoc builds are failing.

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                self?.router.dismiss(viewController)
+            })
+            .flatMap { [interactor] (viewController, params) in
+                interactor
+                    .send(parameters: params)
+                    .map {
+                        viewController
+                    }
+            }
+            .receive(on: RunLoop.main)
```
Comment (szabinst) [inline Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:178]: Instead of hard coding `RunLoop.main` we could inject a `scheduler: AnySchedulerOf<DispatchQueue> = .main` to the view model's initializer and then we'd be able to make the unit tests synchronous by injecting `.immediate`.

File: Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            .sink(receiveCompletion: { completion in
+                if case .failure = completion {
+                    Logger.shared.error("ComposeMessageView message failure")
+                    let title = NSLocalizedString("Failure", comment: "")
+                    let message = NSLocalizedString("Please try again!", comment: "")
+                    self.showResultDialog(title: title, message: message)
+                    self.isSendingMessage = false
+                }
```
Comment (szabinst) [inline Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:186]: This doesn't trigger an alert view for me.

**Review by szabinst — CHANGES_REQUESTED** [review]
I would have a few UX improvements. 

- You can tap on the Course label to select a course, but on the "To" row, you have to select the "+" sign on the far right. 
- I spent a good minute repeatedly tapping on the "Message" label trying to figure out why the keyboard won't show up, when I realized I need to tap the area under it. 
- I tried to trigger the error warning by turning off Wi-Fi before sending a message, but I didn't receive any error alert.

**Review by vargaat — CHANGES_REQUESTED** [review]

**Review by szabinst — APPROVED** [review]
Good job! QA + 1

**Review by vargaat — APPROVED** [review]
QA+1
